src/synthesizer.ts

```diff
@@ -448,3 +651,63 @@ function assertBound<A>(x: A | undefined): asserts x is NonNullable<A> {
         throw new Error('You must call bindStack() first');
     }
 }
+
+/**
+ * Converts the CDK DockerCacheOption to the Pulumi Docker.CacheToArgs
+ */
+function fromCdkCacheTo(cacheTo?: cdk.DockerCacheOption): docker.types.input.CacheToArgs[] | undefined {
+    if (!cacheTo) {
+        return undefined;
+    }
+    return [{ [cacheTo.type]: cacheTo.params }];
+}
+
+/**
+ * Converts the CDK DockerCacheOption to the Pulumi Docker.CacheFromArgs
+ */
+function fromCdkCacheFrom(cacheFrom?: cdk.DockerCacheOption[]): docker.types.input.CacheFromArgs[] | undefined {
+    if (!cacheFrom) {
+        return undefined;
+    }
+    return cacheFrom.map((cache) => {
+        return {
+            [cache.type]: cache.params,
+        };
+    });
+}
+
+/**
+ * Converts the CDK NetworkMode to the Pulumi Docker.NetworkMode
+ *
+ * @param networkMode - The cdk network mode
+ * @returns The docker network mode
+ */
+export function asNetworkMode(networkMode?: string): docker.NetworkMode | undefined {
+    if (networkMode === undefined) {
+        return undefined;
+    }
+
+    // docker.NetworkMode is a `type` and I can't find a way to compare a string value to a type
```

Comment:
Doesn't `Object.values(NetworkMode)` work to retrieve those values here: https://github.com/pulumi/pulumi-docker-build/blob/4c030122685eaebbef6ac58494f1443f75640cd5/sdk/nodejs/types/enums/index.ts#L35-L48?

It might be a newer language feature though.

Reply:
I got it to work by casting the result of `Object.values(NetworkMode) as string[]`